src/unix/sunos.c

```diff
@@ -475,6 +474,12 @@ static void uv__fs_event_read(uv_loop_t* loop,
     handle = (uv_fs_event_t*) pe.portev_user;
     assert((r == 0) && "unexpected port_get() error");
 
+    if (uv__is_closing(handle)) {
+      uv__handle_stop(handle);
+      uv__make_close_pending((uv_handle_t*) handle);
+      break;
+    }
```

Comment:
In reviewing #3620, I noticed a possible defect introduced here, and wanted to check my understanding. Doesn't this need to now check if the user attempted to stop the handle here?
```suggestion
    } else if (handle->fd == PORT_DELETED) {
      uv__handle_stop(handle);
      break;
    }
```

Reply:
Secondly, there is a `uv__is_active` check in `uv_fs_event_start`, and I think that is now incorrect (or at least insufficient) and also needs to permit the case where `uv_fs_event_stop` didn't finish yet. But also while maintaining this fix (so that if you do a sequence of stop/event/start/close, it will still remember to defer the close).

I am mildly tempted to suggest we should instead soft-deprecate `uv_fs_event_stop` (in favor of `uv_close`)

Reply:
It's a while since I looked at this now but, yes, I think you're right for the `us_fs_event_stop()` case when the event has already been despatched and is awaiting delivery. It will need a `uv__make_close_pending()` too though, wouldn't it? It could then just be added to the condition above:

```
    if (uv__is_closing(handle) || handle->fd == PORT_DELETED) {
...
```

Reply:
`uv__make_close_pending` shouldn't be triggered, since the handle is only paused, not closed